### PR TITLE
update to nette 3.0

### PR DESCRIPTION
**Pull request by slepic**
I have updated composer requirement of nette to 3.0 and php to 7.0 (needed by nette anyway). At this point I have fixed every error reported by running `composer test`, except for those cases mentioned in the TODO below (I have actualy found ways to modify vendor packages to make those problems go away, but to fix it properly it has to be done on the grido side, but I havent found those ways yet.).
Let me first ask, are you willing to merge this (when completed) and release a new version of grido compatible with nette 3.0. Or would you prefer if someone else maintained this package, so I better off with my fork?

TODO:
 - ~~for some reason, in tests, RouteList contains itself as a route, resulting in infinite loop~~ (not grido issue).
 - ~~in tests, nette often thinks that there is a csrf attack and makes redirection instead of handlig the requested action. (looks like missing `@crossOrigin` annotation, or bad sameSite detection)~~
- dont use dev-master versions (kdyby packages) in composer.json

### CODE REVIEW

**Comment by o5** [comment]
@slepic I've no problem with merge with green Travis.

**Comment by slepic** [comment]
ok so everything should be fine now, except these kdyby packages dont have stable release for nette 3.0:
- kdyby/console
- kdyby/doctrine
- kdyby/events

so they are on dev-master

and the out of memory problem coming from kdyby/console infinite loop in router is solved in https://github.com/Kdyby/Console/pull/85
so once they merge this, the build of this repo should be fixed as well.

There is also a workaround for this problem:
https://github.com/Kdyby/DoctrineCache/issues/23
so once that one is solved, the workaround could be removed.
https://github.com/o5/grido/commit/066c9f56381be0382a230288e1a9393fde215a8d#diff-2e76b9f7fcf72fd68fffc5fb11681690R109

**Comment by slepic** [comment]
@o5 The kdyby/console issue is merged. Can you please rerun travis build to see if everything works now? Seems I have no right to do so (unless i'm just blind)

**Comment by o5** [comment]
@slepic build has been [restarted](https://travis-ci.org/o5/grido/builds/589371634) but still fails :/

**Comment by slepic** [comment]
@o5 ok, so 
- the incompatibility of implementation of Nette\Http\IResponse i didnt figure earlier because i was testing on PHP 7.2 where it is allowed (parameter type widening, https://www.php.net/manual/en/migration72.new-features.php), but the travis runs in 7.1 where this is not allowed.
- problem with operation returning ids as ints instead of string is caused by nette (https://github.com/nette/forms/blob/v3.0.1/src/Forms/Container.php#L117) in combination with BC break in PHP 7.2 (Convert numeric keys in object and array casts , https://www.php.net/manual/en/migration72.incompatible.php). Fixed this by making sure that the keys are always returned as strings from inside grido's Operation class no matter what is obtained from nette.

**Comment by o5** [comment]
@slepic Sorry for delay. Could you please polish your commits?

**Comment by slepic** [comment]
@o5 Could you be more specific, please? :)

**Comment by o5** [comment]
I meant squash what is possible, better commit message (without `travis vol 2`, `wip`, etc)

**Comment by slepic** [comment]
@o5 Oh sorry, but I don't want to or even know how to mess up with commit history... Can't you just do it on merge? Github can handle this for you. Just squash it into one commit and let the message be `nette 3.0 support` or alike...

**Comment by sasule** [comment]
@slepic Here is a nice tutorial on how to make a squash: https://stackoverflow.com/questions/5189560/squash-my-last-x-commits-together-using-git
When You are in squash process then you see all messages, you can delete them and write a new one which will be used.

When you are finished with work on new commit message you have to force push to the repository.

**Comment by slepic** [comment]
@sasule Thanks, I would have eventualy figurde it out. I just really dont like to make changes to already published commits.

@o5 so it's squashed into one commit now. The individual commits didn't really have their own isolated scope....

File: .travis.yml [hunk 1]
```diff
@@ -21,4 +20,4 @@ script:
     - vendor/bin/tester -p php $COVERAGE tests/
 
 after_script:
-    - if [ $TRAVIS_PHP_VERSION = '5.6' ]; then wget https://scrutinizer-ci.com/ocular.phar && php ocular.phar code-coverage:upload --format=php-clover coverage.xml; fi
+    - if [ $TRAVIS_PHP_VERSION = '5.6' ]; then wget https://scrutinizer-ci.com/ocular.phar && php ocular.phar code-coverage:upload --format=php-clover coverage.xml; fi
```
Comment (o5) [inline .travis.yml]: This condition is always false now and project will be without code-coverage report.

**Comment by o5** [comment]
@slepic Change looks good, but there is something in [Travis log](https://travis-ci.org/o5/grido/jobs/608629273?utm_medium=notification&utm_source=github_status):

```
Notifying that no code coverage data is available for repository "g/o5/grido" and revision "ac3dac873010b07d28b13723dfeea35c48043ea4"... Done
```

Scrutinizer [reports](https://scrutinizer-ci.com/g/o5/grido/inspections/51aeca2c-d851-48f3-8dcf-ac18a6bcf87f) `Waiting for External Code Coverage`.

**Comment by slepic** [comment]
@o5 Do you have any idea what that means? Seems to me it should be ok.

**Comment by o5** [comment]
@slepic Unfortunately don't in this time.

**Comment by o5** [comment]
Merged https://github.com/o5/grido/commit/7fbb26385519019b1e7ade25c320be5f4da2745f

**Comment by slepic** [comment]
@o5 A bit late, but anyway, thank you very much :) Could I yet ask you to tag a 3.0.0 release? We are already using the master head on one application in production and another one is about to be released, but I don't wanna keep it on dev-master, although we keep our packages locked...

**Comment by o5** [comment]
@slepic https://github.com/o5/grido/releases/tag/3.1.0

**Pull request closed without merging**